### ISSUE TITLE
Move to SmallRye

### ISSUE THREAD

**Issue by Ladicek** [wildfly/jandex#124]
For the 3.0 release, we want to move Jandex to SmallRye (and use the automated release process SmallRye has). Ideally, all Jandex repos (this one, the `typeannotation-test`, and the Maven plugin) would be brought into one (covered by #122).

The `groupId` would change from `org.jboss[.jandex]` to `io.smallrye` (or `io.smallrye.jandex`, but I'd probably prefer the first one :-) ). The `artifactId`s or packages wouldn't change (so packages would still be `org.jboss.jandex`), as that would be too big of a break.

**Comment by darranl** [comment]
I don't think everyone will get the individual notifications from GitHub, maybe worth forwarding to the wildfly-dev list?

**Comment by Ladicek** [comment]
I agree, but I also think we should do that after the 2.4.0.Final release. I mean -- what I just added to the 2.4.0.Final milestone was originally referred to as 3.0. Then we decided we should call it 2.4, as there's nothing breaking, and do a 3.0 with some breakages that are needed (including this one). I think that makes most sense, but until that is actually done, I'd like to leave the door open :-)

**Comment by darranl** [comment]
+1 whenever makes more sense.

**Comment by gastaldi** [comment]
@Ladicek moving to Smallrye sounds like a wonderful idea, +1 to that. About the package name, I wonder if that wouldn't be an opportunity to introduce some major API changes (eg. introduce a new API under `io.smallrye.jandex` and keep the original API under `org.jboss.jandex`)?

**Comment by Ladicek** [comment]
I'm not sure we want to have 2 APIs. Also current implementation relies heavily on package-private methods, which precludes such split. I think we'll just live with some breaking changes :-)

**Comment by Ladicek** [comment]
All the changes required in the code base (`groupId` and parent POM) are now done on the `smallrye` branch. I'm not closing this issue because Jandex is still under the `wildfly` org. I believe we should transfer this repo to the `smallrye` org when the 3.0 release is almost ready. In the meantime, the changes for the 3.0 release should be done as PRs against the `smallrye` branch.

**Comment by Ladicek** [comment]
I just moved the Jandex repository to the `smallrye` GitHub org in preparation for a 3.0.0-RC1 release. Hence this is done for good.

**Comment by jonathanlermitage** [comment]
If my understanding is correct, we can consider the `org.jboss.jandex` group-id moved to `io.smallrye.jandex`?  
I'm working on [Maven oga plugin](https://github.com/jonathanlermitage/oga-maven-plugin), a plugin that alerts developers when an artifact moved to new coordinates, but the affected pom files do not offer the relocation information. This is based on a huge list of [coordinates migrations](https://github.com/jonathanlermitage/oga-maven-plugin/blob/master/uc/og-definitions.json) and, I would like to know your opinion before adding the migration rule(s) for Jandex. Thanks!

**Comment by Ladicek** [comment]
> If my understanding is correct, we can consider the `org.jboss.jandex` group-id moved to `io.smallrye.jandex`?

No. The artifacts are moved like this:

`org.jboss:jandex` -> `io.smallrye:jandex`
`org.jboss.jandex:jandex-maven-plugin` -> `io.smallrye:jandex-maven-plugin`

See https://smallrye.io/blog/jandex-3-0-0/ for more.

> I'm working on [Maven oga plugin](https://github.com/jonathanlermitage/oga-maven-plugin), a plugin that alerts developers when an artifact moved to new coordinates, but the affected pom files do not offer the relocation information.

We do have relocations in place for `org.jboss:jandex`. Does it not work?

> This is based on a huge list of [coordinates migrations](https://github.com/jonathanlermitage/oga-maven-plugin/blob/master/uc/og-definitions.json) and, I would like to know your opinion before adding the migration rule(s) for Jandex. 

See above.

Thanks for reaching out, please let me know if you need more information!
